src/os/bluestore/bluestore_tool.cc

```diff
@@ -219,6 +227,59 @@ void inferring_bluefs_devices(vector<string>& devs, std::string& path)
   }
 }
 
+void bluefs_import(
+  string input_file,
+  string dest_file,
+  CephContext *cct,
+  const string& path,
+  const vector<string>& devs)
+{
+  int r;
+  std::ifstream f(input_file.c_str(), std::ifstream::binary);
+  if (!f) {
+    r = -errno;
+    cerr << "open " << input_file.c_str() << " failed: " << cpp_strerror(r) << std::endl;
+    exit(EXIT_FAILURE);
+  }
+
+  std::unique_ptr<BlueFS> bs = std::unique_ptr<BlueFS>(open_bluefs(cct, path, devs));
+
+  BlueFS::FileWriter *h;
+  fs::path file_path(dest_file);
+  const string dir = file_path.parent_path();
+  const string file_name = file_path.filename();
+  r = bs->open_for_write(dir, file_name, &h, true);
```

Comment:
You can simplify code below by providing overwrite=false param here.
This way BlueFS resets existing file if any and there is no need to do manual truncate..

Reply:
Ok, I optimize it,
thank you.